x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/AbstractFunctionTestCase.java

```diff
@@ -470,7 +471,9 @@ public final void testSimpleWithNulls() { // TODO replace this with nulls insert
             // function processing of the first parameter(s) could raise an exception/warning. (But hasn't been the case so far.)
             // For n-ary functions, dealing with one multivalue (before hitting the null parameter injected above) will now trigger
             // a warning ("SV-function encountered a MV") that thus needs to be checked.
-            if (simpleData.stream().anyMatch(List.class::isInstance) && testCase.getExpectedWarnings() != null) {
+            if (this instanceof AbstractMultivalueFunctionTestCase == false
```

Comment:
Maybe I'm just confused, but I feel like this check is contradictory to the comment above it.  Does the comment need to be rewritten? or just added to?